src/data/set/basic.lean

```diff
@@ -1722,6 +1722,42 @@ theorem subsingleton_of_image {α β : Type*} {f : α → β} (hf : function.inj
 theorem univ_eq_true_false : univ = ({true, false} : set Prop) :=
 eq.symm $ eq_univ_of_forall $ classical.cases (by simp) (by simp)
 
+section preorder
+
+variables [preorder α] [preorder β] (f : α → β)
+
+/-! ### Monotonicity on singletons -/
+
+protected lemma subsingleton.monotone_on (h : s.subsingleton) :
+  monotone_on f s :=
+λ a ha b hb _, (congr_arg _ (h ha hb)).le
+
+protected lemma subsingleton.antitone_on (h : s.subsingleton) :
+  antitone_on f s :=
+λ a ha b hb _, (congr_arg _ (h hb ha)).le
+
+protected lemma subsingleton.strict_mono_on (h : s.subsingleton) :
+  strict_mono_on f s :=
+λ a ha b hb hlt, (hlt.ne (h ha hb)).elim
+
+@[simp] protected lemma subsingleton.strict_anti_on (h : s.subsingleton) :
```

Comment:
Why is this one simp but not the three similar ones above?

Reply:
I don't think those should be simp.

Reply:
Theres a simp there cause I copied Yael's code from [here](https://github.com/leanprover-community/mathlib/pull/14896#discussion_r906722837). I've removed it now